Assets/Scripts/Layers/LayerTypes/LayerNL3DBase.cs

```diff
@@ -91,31 +76,39 @@ public bool ActiveInHierarchy
             }
         }
 
-        public readonly UnityEvent<string> NameChanged = new();
-        public readonly UnityEvent<bool> LayerActiveInHierarchyChanged = new();
-        public readonly UnityEvent<Color> ColorChanged = new();
-        public readonly UnityEvent LayerDestroyed = new();
+        [JsonIgnore] public readonly UnityEvent<string> NameChanged = new();
+        [JsonIgnore] public readonly UnityEvent<bool> LayerActiveInHierarchyChanged = new();
+        [JsonIgnore] public readonly UnityEvent<Color> ColorChanged = new();
+        [JsonIgnore] public readonly UnityEvent LayerDestroyed = new();
 
-        public readonly UnityEvent<LayerNL3DBase> LayerSelected = new();
-        public readonly UnityEvent<LayerNL3DBase> LayerDeselected = new();
+        [JsonIgnore] public readonly UnityEvent<LayerNL3DBase> LayerSelected = new();
+        [JsonIgnore] public readonly UnityEvent<LayerNL3DBase> LayerDeselected = new();
 
-        public readonly UnityEvent ParentChanged = new();
-        public readonly UnityEvent ChildrenChanged = new();
-        public readonly UnityEvent<int> ParentOrSiblingIndexChanged = new();
+        [JsonIgnore] public readonly UnityEvent ParentChanged = new();
+        [JsonIgnore] public readonly UnityEvent ChildrenChanged = new();
+        [JsonIgnore] public readonly UnityEvent<int> ParentOrSiblingIndexChanged = new();
+
+        public void InitializeParent(LayerNL3DBase initialParent = null)
```

Comment:
is public the best way? This sounds like overexposing the workings of this class; or should this method be called setParent?

Reply:
Or be a property?

Reply:
I wanted to make this protected/private, but then RootLayer cannot access it. It cannot be a property because this function exists to set the initial parent fields after loading since these are not serialized to avoid a circular reference. If you have a suggestion on how to fix/remove this function that would be a great improvement!

Reply:
I'll give it a look in after reading through your comments; but perhaps we should do as you discussed with Sam earlier and do a refactoring round after delivering the feature